code/src/components/Header.js

```diff
@@ -0,0 +1,70 @@
+import React from "react";
+import { useSelector, useDispatch } from "react-redux";
+import moment from "moment";
+import styled from "styled-components";
+import { tasks } from "../reducers/tasks";
+
+export const Header = () => {
+  const taskStore = useSelector(store => store.tasks.tasks);
```

Comment:
Since you're only using the length of the array, I'd recommend returning that directly from the selector:

```
const numberOfTasks = useSelector(store => store.tasks.tasks.length)
```

When you do it like you have, this `Header` component will re-render whenever the array of tasks changes (for example toggling a task's completion), but if you return the length from the selector, it'll only re-render when the length of the array changes. Small optimisation, but good to keep in the back of your head if you're working on a bigger project where you want to avoid unnecessary re-renders.